### PR TITLE
Avatar cache tweaks & hook update

### PR DESCRIPTION
**Pull request by greenish**
this should fix 
- #1906
- #1888
- #1917

And a potential fix for #1870 -> I cant reproduce this issue, my theory is that specific NFTs cause this.

### CODE REVIEW

File: packages/themes/src/index.tsx [hunk 1]
```diff
@@ -318,7 +318,7 @@ export const lightTheme: Partial<Theme> & { custom: { colors: CustomColors } } =
         secondary: LIGHT_FONT_COLOR_1,
         primaryButton: LIGHT_BRAND_COLOR,
         primaryButtonTextColor: LIGHT_BACKGROUND_COLOR_1,
-        secondaryButton: LIGHT_BACKGROUND_COLOR_1,
+        secondaryButton: LIGHT_BACKGROUND_COLOR_2,
```
Comment (armaniferrante) [inline packages/themes/src/index.tsx:321]: Seems like this might regress colors elsewhere.

Comment (armaniferrante) [reply packages/themes/src/index.tsx:321]: Confirmed. For example, the cancel button on the send NFT screen.

Comment (armaniferrante) [inline packages/themes/src/index.tsx:321]:
```suggestion
        secondaryButton: LIGHT_BACKGROUND_COLOR_1,
```

**Comment by armaniferrante** [comment]
Colors look good even after merging.

**Comment by armaniferrante** [comment]
Note that this did not solve https://github.com/coral-xyz/backpack/issues/1888.

**Comment by greenish** [comment]
> Note that this did not solve #1888.

Ah this should have been #1889 

GH autocomplete very easily messes up

**Pull request merged**